spec/controllers/subscriptions_controller_spec.rb

```diff
@@ -146,6 +146,35 @@
         end
       end
 
+      context "when subscription is a gift" do
+        let(:gifter) { create(:user) }
+        let(:giftee) { create(:user) }
+        let(:product) { create(:membership_product, user: seller) }
+        let(:gifter_purchase) { create(:purchase, :gift_sender, link: product, purchaser: gifter, is_original_subscription_purchase: true) }
+        let(:giftee_purchase) { create(:purchase, :gift_receiver, link: product, purchaser: giftee) }
+        let!(:gift) { create(:gift, gifter_purchase:, giftee_purchase:, link: product) }
+        let!(:gifted_subscription) { create(:subscription, link: product, user: giftee) }
+
+        before do
+          gifter_purchase.update!(subscription: gifted_subscription, gift_given: gift)
+          giftee_purchase.update!(subscription: gifted_subscription, gift_received: gift)
```

Comment:
Any reason why we can't assign these attributes in the `let` statement a couple of lines above?